### PR TITLE
[icu] Manual roll of icu

### PR DESCRIPTION
**Pull request by filmil**
Manual roll of the ICU library. This normally happens automatically, but this
time around, ICU folks changed their license, so that needs to be fixed up.
I wish they would stop doing that. :)

This manual change should enable us to turn the auto-roller back on.

https://github.com/flutter/flutter/issues/149684

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by filmil** [comment]
@matanlurey Please review.

(if all passes, I will also need you to merge this, as I'm not a committer.)

**Comment by filmil** [comment]
Can you please merge this for me? I am not a committer.

**Review by matanlurey — APPROVED** [review]

**Comment by matanlurey** [comment]
> Can you please merge this for me? I am not a committer.

I've added `autosubmit`, as soon as the tree turns green it should auto-merge. Thanks for the fix!

**Comment by ChristianEdwardPadilla** [comment]
The changes here seem to be causing license headers to fail to be identified.

See license compliance presubmit result here: cl/641003589.

Can this be rolled back?

**Comment by filmil** [comment]
> cl/641003589

Isn't this something that should be fixed downstream?

**Comment by filmil** [comment]
To wit, ICU license *did* change in ICU release `74.1`. So if downstream has a rigid dependency on the ICU license not changing, well, then that check will need to be adapted.

**Comment by matanlurey** [comment]
@filmil I've engaged Christian in an internal chat - I also don't understand what we expect y'all to have done differently.

I'll come back here if we need any changes (unlikely)

**Comment by filmil** [comment]
> @filmil I've engaged Christian in an internal chat - I also don't understand what we expect y'all to have done differently.

If I may venture a guess, I think downstream made a reasonable assumption that the upstream license wouldn't change.  But it did.
This was my comment all the way at the top of this PR, **emphasis** mine:

> Manual roll of the ICU library. This normally happens automatically, but this
> time around, ICU folks changed their license, so that needs to be fixed up.
> **I wish they would stop doing that. :) ** [...]

**Comment by matanlurey** [comment]
LOL. Thank you for that bit of humor :)

**Comment by matanlurey** [comment]
The change was approved internally, no action required.

**Pull request merged**
